src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageSourceMappingActionViewModel.cs

```diff
@@ -125,9 +130,16 @@ public void SettingsChanged()
             RaisePropertyChanged(nameof(CanAutomaticallyCreateSourceMapping));
         }
 
-        public static PackageSourceMappingActionViewModel Create(INuGetUI uiController)
+        public void Dispose()
         {
-            return new PackageSourceMappingActionViewModel(uiController);
+            _settings.SettingsChanged -= Settings_SettingsChanged;
```

Comment:
Can you confirm that this `Dispose` is explicitly called somewhere, and it's not relying on the GC calling dispose? As per my previous comment, `_settings.SettingsChanged` will hold a reference to this object, hence the GC will never dispose this object until after `_settings` releases the instance (the event is unsubscribed).

I had a look through the PR, but I don't see any new `.Dispose()` calls,